### PR TITLE
Adding the homebrew formula to Chapel util directory.   

### PR DESCRIPTION
**Pull request by tzinsky**
This would be considered a mirror of what is in the Homebrew project.  The PR is not needed for the 1.26 release.   It adds a packaging directory and lands the homebrew files there with a mirror (chapel-release.rb), the formula that would be used with the next chapel release(chapel-main.rb), and a readme.md covering how to build locally.

### CODE REVIEW

**Comment by mppf** [comment]
Since this is just adding the existing 1.25.1 formula, I think the main question is where to put it. I wonder if we should add some place that makes sense to collect files to support Chapel packages on different systems. Arguably we have already util/buildRelease and util/build_configs and util/dockerfiles that do this so I can understand why it is a sibling there. @bradcray do you have a suggestion here?

**Comment by bradcray** [comment]
>  @bradcray do you have a suggestion here?

I tend to think that the name of the new directory added here, `util/formula`, is a bit vague ("formula for what?") and single-purpose ("the homebrew formula only").  If we need/want this to be in before the release, the simplest change that would make me happy would be to rename the directory to `util/homebrew/` or to push it down a level under `util/buildRelease/homebrew`.  I also like the idea of having `util/packaging/homebrew/`, but would then probably want to push docker under there as well, reconsider the relationship between `buildRelease` and `packaging`, etc. and that feels like more than we should take on right before the code freeze.

If this doesn't need to go in before the release, then I agree that it'd be nice to have a setup in which docker and homebrew were siblings under some common directory, whatever that is and think that's worth additional thought and discussion.

File: util/homebrew/chapel-main.rb [hunk 1]
```diff
@@ -0,0 +1,430 @@
+class Chapel < Formula
+  desc "Programming language for productive parallel computing at scale"
+  homepage "https://chapel-lang.org/"
+  url "https://github.com/chapel-lang/chapel/releases/download/1.25.1/chapel-1.25.1.tar.gz"
```
Comment (mppf) [inline util/homebrew/chapel-main.rb:4]: I would expect this to be `chapel-release.rb` and the other to be `chapel-main` ? Anyway I am currently thinking it is best for this to be post-release; in that event, we can merge the Homebrew formula we end up with for 1.26 as chapel-release.rb.

**Comment by tzinsky** [comment]
I have updated the PR to rename the directory but I can move it to better align with the suggestion about packaging.  Also, it is not urgent for the release and we have a working formula.   I will update this PR with the suggested changes and we can move on from there...

**Pull request merged**